### PR TITLE
Search index EE ranker tests

### PR DESCRIPTION
**Pull request by qnkhuat**
refs #50261

### CODE REVIEW

File: enterprise/backend/test/metabase_enterprise/search/scoring_test.clj [hunk 1]
```diff
@@ -149,3 +150,33 @@
...
+     {:model "collection" :id 2 :name "collection official" :official_collection true}]
+    (testing "official collections has higher rank"
+      (mt/with-premium-features #{:official-collections}
+        (is (= [["collection" 2 "collection official"]
+                ["collection" 1 "collection normal"]]
+               (fulltext.scoring-test/search :official-collection "collection")))))
+    (testing "only if feature is enabled"
+      (mt/with-premium-features #{}
+        (is (= [["collection" 1 "collection normal"]
+                ["collection" 2 "collection official"]]
```
Comment (crisptrutski) [inline enterprise/backend/test/metabase_enterprise/search/scoring_test.clj:166]: These are more cases where we're relying on search being stable. It probably won't be with other backends.

I guess we can cross that bridge when we get to it, I don't have a good solution besides trading off against another weaker weight, which creates ranker coupling in these "unit" tests.

Why not keep the same approach as for the non-EE tests, and just assert that the order is different? Then in that helper we can mention this future issue, and save our future selves some head scratching.

File: test/metabase/search/fulltext/scoring_test.clj [hunk 2]
```diff
@@ -73,18 +72,17 @@
                         :models           search.config/all-models
                         :model-ancestors? false}
                        raw-ctx))]
-      (is (get (search.scoring/scorers search-ctx) ranker-key) "The ranker is enabled")
```
Comment (crisptrutski) [inline test/metabase/search/fulltext/scoring_test.clj:76]: What's your reason for removing this?

Comment (qnkhuat) [reply test/metabase/search/fulltext/scoring_test.clj:76]: yea, I should have put some context here; I wanted to test that search. There is no feature flag. if we have this the test will fail.

**Review by crisptrutski — APPROVED** [review]

**Comment by crisptrutski** [comment]
There are some small improvements I think we can make, but I'll just do as a fast follow since you've moved onto ingestion tests.

**Comment by crisptrutski** [comment]
Oh snap, more conflicts.

File: test/metabase/search/fulltext/scoring_test.clj [hunk 3]
```diff
@@ -31,19 +31,17 @@
...
 
 (defn search-results
   "Like search-results* but with a sanity check that search without weights returns a different result."
   [ranker-key search-string & {:as raw-ctx}]
-  (let [result (search-results* search-string raw-ctx)]
-    (is (not= (search-no-weights ranker-key search-string raw-ctx)
-              result)
+  (let [result   (with-weights {ranker-key 1} (search-results* search-string raw-ctx))
+        inverted (with-weights {ranker-key -1} (search-results* search-string raw-ctx))]
+    ;; note that this may not be a strict reversal, due to tries.
```
Comment (crisptrutski) [inline test/metabase/search/fulltext/scoring_test.clj:43]:
```suggestion
    ;; note that this may not be a strict reversal, due to ties.
```

**Pull request merged**